src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -608,6 +608,12 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
 
         internal BoundLocalDeconstructionDeclaration BindDeconstructionDeclarationStatement(DeconstructionDeclarationStatementSyntax node, DiagnosticBag diagnostics)
         {
+            var constKeyword = node.Modifiers.FirstOrDefault(SyntaxKind.ConstKeyword);
+            if (constKeyword != default(SyntaxToken))
```

Comment:
@cston Problem with putting the squiggle on the type is there might not be "a" type, for example in
```cs
const (int x, int y) = e;
```
or, more generally, once we support wildcards
```cs
const (*, *) = e;
```
I think it could be the modifier `const`, as in the PR, or the deconstruction, or even the whole statement.